mypy/subtypes.py

```diff
@@ -143,10 +143,15 @@ def visit_instance(self, left: Instance) -> bool:
             rname = right.type.fullname()
             if not left.type.has_base(rname) and rname != 'builtins.object':
                 return False
-
             # Map left type to corresponding right instances.
             t = map_instance_to_supertype(left, right.type)
-
+            # Check that one of the types does not have type args or
+            # the number of type args is the same and
+            # each left type arg is acceptable in place of the matching right one
+            if not t.args or not right.args:
+                return True
+            if len(t.args) != len(right.args):
+                return False
```

Comment:
I can do that, but isn't it wrong for `visit_instances` to return `True` when the two types have a different number of type parameters? `Tuple[int]` vs `Tuple[int, str]` is one example (which caused the crash), but in the future (or even now, though I can't think of any) there may be more cases where two types have different number of type parameters. Shouldn't we, generally speaking, refuse to accept `X[A]` in place of `X[B, C]` or vice versa, when `A`, `B`,  `C` are not `Any`?